org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/cleanup/LambdaExpressionAndMethodRefCleanUp.java

```diff
@@ -0,0 +1,57 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Red Hat Inc. and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Red Hat Inc. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.ls.core.internal.cleanup;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.jdt.core.dom.CompilationUnit;
+import org.eclipse.jdt.core.manipulation.CleanUpContextCore;
+import org.eclipse.jdt.core.manipulation.ICleanUpFixCore;
+import org.eclipse.jdt.internal.corext.fix.LambdaExpressionAndMethodRefFixCore;
+
+/**
+ * Represents a cleanup that does several actions to clean up lambda expression
+ */
+public class LambdaExpressionAndMethodRefCleanUp implements ISimpleCleanUp {
+
+	/* (non-Javadoc)
+	 * @see org.eclipse.jdt.ls.core.internal.cleanup.ISimpleCleanUp#getIdentifier()
+	 */
+	@Override
+	public String getIdentifier() {
+		return "lambdaExpressionAndMethodRef";
```

Comment:
Since this new clean up is more general, should we call the new one `lambdaExpression` and rename the other to something like `lambdaExpressionFromAnonymousClass`?

Reply:
Yeah that's fine with me. Only downside to that is if anyone is using that as an entry in their cleanup list, they'd have to switch over, but it's a fairly minor change. We would just have to adjust, for example the settings on the client side (eg. vscode-java)